commit 4ead2b84da81e3283107ab0cf58598fb8bcecf3a
Author: Bryant G. Ly <bryantly@linux.vnet.ibm.com>
Date:   Tue May 9 11:50:26 2017 -0500

    ibmvscsis: Clear left-over abort_cmd pointers
    
    commit 98883f1b5415ea9dce60d5178877d15f4faa10b8 upstream.
    
    With the addition of ibmvscsis->abort_cmd pointer within
    commit 25e78531268e ("ibmvscsis: Do not send aborted task response"),
    make sure to explicitly NULL these pointers when clearing
    DELAY_SEND flag.
    
    Do this for two cases, when getting the new new ibmvscsis
    descriptor in ibmvscsis_get_free_cmd() and before posting
    the response completion in ibmvscsis_send_messages().
    
    Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>
    Reviewed-by: Michael Cyr <mikecyr@linux.vnet.ibm.com>
    Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/scsi/ibmvscsi_tgt/ibmvscsi_tgt.c b/drivers/scsi/ibmvscsi_tgt/ibmvscsi_tgt.c
index 904422f..ad36e51 100644
--- a/drivers/scsi/ibmvscsi_tgt/ibmvscsi_tgt.c
+++ b/drivers/scsi/ibmvscsi_tgt/ibmvscsi_tgt.c
@@ -1169,6 +1169,8 @@ static struct ibmvscsis_cmd *ibmvscsis_get_free_cmd(struct scsi_info *vscsi)
 		cmd = list_first_entry_or_null(&vscsi->free_cmd,
 					       struct ibmvscsis_cmd, list);
 		if (cmd) {
+			if (cmd->abort_cmd)
+				cmd->abort_cmd = NULL;
 			cmd->flags &= ~(DELAY_SEND);
 			list_del(&cmd->list);
 			cmd->iue = iue;
@@ -1773,6 +1775,7 @@ static void ibmvscsis_send_messages(struct scsi_info *vscsi)
 				if (cmd->abort_cmd) {
 					retry = true;
 					cmd->abort_cmd->flags &= ~(DELAY_SEND);
+					cmd->abort_cmd = NULL;
 				}
 
 				/*
